### PR TITLE
uniswap v3 swap fix

### PR DESCRIPTION
**Pull request by pradyuman-verma**
- isEth logic fixed

### CODE REVIEW

File: contracts/arbitrum/connectors/uniswap/v3_swap/helpers.sol [hunk 1]
```diff
@@ -70,8 +70,8 @@ abstract contract Helpers is DSMath, Basic {
 		uint256 _sellAmt = swapRouter.exactOutputSingle(params);
 		require(_slippageAmt >= _sellAmt, "Too much slippage");
 
-		isEth = address(_buyAddr) == wethAddr;
-		convertWethToEth(isEth, _buyAddr, _buyAmt);
+		isEth = address(buyData.buyAddr) == wethAddr;
+		convertEthToWeth(isEth, _buyAddr, _slippageAmt);
```
Comment (thrilok209) [inline contracts/arbitrum/connectors/uniswap/v3_swap/helpers.sol]:
```suggestion
		convertWethToEth(isEth, _buyAddr, _buyAmt);
```

File: contracts/arbitrum/connectors/uniswap/v3_swap/helpers.sol [hunk 2]
```diff
@@ -131,8 +131,8 @@ abstract contract Helpers is DSMath, Basic {
 		uint256 _buyAmt = swapRouter.exactInputSingle(params);
 		require(_slippageAmt <= _buyAmt, "Too much slippage");
 
-		isEth = address(_buyAddr) == wethAddr;
-		convertWethToEth(isEth, _buyAddr, _buyAmt);
+		isEth = address(sellData.buyAddr) == wethAddr;
+		convertEthToWeth(isEth, _buyAddr, _sellAmt);
```
Comment (thrilok209) [inline contracts/arbitrum/connectors/uniswap/v3_swap/helpers.sol]:
```suggestion
		convertWethToEth(isEth, _buyAddr, _sellAmt);
```

File: contracts/mainnet/connectors/uniswap/v3_swap/helpers.sol [hunk 3]
```diff
@@ -69,8 +69,8 @@ abstract contract Helpers is DSMath, Basic {
 		uint256 _sellAmt = swapRouter.exactOutputSingle(params);
 		require(_slippageAmt >= _sellAmt, "Too much slippage");
 
-		isEth = address(_buyAddr) == wethAddr;
-		convertWethToEth(isEth, _buyAddr, _buyAmt);
+		isEth = address(buyData.buyAddr) == wethAddr;
+		convertEthToWeth(isEth, _buyAddr, _slippageAmt);
```
Comment (thrilok209) [inline contracts/mainnet/connectors/uniswap/v3_swap/helpers.sol]:
```suggestion
		convertWethToEth(isEth, _buyAddr, _buyAmt);
```

File: contracts/polygon/connectors/uniswap/v3_swap/helpers.sol [hunk 4]
```diff
@@ -69,8 +69,8 @@ abstract contract Helpers is DSMath, Basic {
 		uint256 _sellAmt = swapRouter.exactOutputSingle(params);
 		require(_slippageAmt >= _sellAmt, "Too much slippage");
 
-		isMatic = address(_buyAddr) == wmaticAddr;
-		convertWmaticToMatic(isMatic, _buyAddr, _buyAmt);
+		isMatic = address(buyData.buyAddr) == maticAddr;
+		convertMaticToWmatic(isMatic, _buyAddr, _slippageAmt);
```
Comment (thrilok209) [inline contracts/polygon/connectors/uniswap/v3_swap/helpers.sol]:
```suggestion
		convertWmaticToMatic(isMatic, _buyAddr, _buyAmt);
```

File: contracts/optimism/connectors/uniswap/v3_swap/helpers.sol [hunk 5]
```diff
@@ -131,8 +131,8 @@ abstract contract Helpers is DSMath, Basic {
 		uint256 _buyAmt = swapRouter.exactInputSingle(params);
 		require(_slippageAmt <= _buyAmt, "Too much slippage");
 
-		isEth = address(_buyAddr) == wethAddr;
-		convertWethToEth(isEth, _buyAddr, _buyAmt);
+		isEth = address(sellData.buyAddr) == ethAddr;
+		convertEthToWeth(isEth, _buyAddr, _sellAmt);
```
Comment (thrilok209) [inline contracts/optimism/connectors/uniswap/v3_swap/helpers.sol]:
```suggestion
		convertWethToEth(isEth, _buyAddr, _buyAmt);
```

File: contracts/optimism/connectors/uniswap/v3_swap/helpers.sol [hunk 6]
```diff
@@ -70,8 +70,8 @@ abstract contract Helpers is DSMath, Basic {
 		uint256 _sellAmt = swapRouter.exactOutputSingle(params);
 		require(_slippageAmt >= _sellAmt, "Too much slippage");
 
-		isEth = address(_buyAddr) == wethAddr;
-		convertWethToEth(isEth, _buyAddr, _buyAmt);
+		isEth = address(buyData.buyAddr) == ethAddr;
+		convertEthToWeth(isEth, _buyAddr, _slippageAmt);
```
Comment (thrilok209) [inline contracts/optimism/connectors/uniswap/v3_swap/helpers.sol]:
```suggestion
		convertWethToEth(isEth, _buyAddr, _buyAmt);
```

File: contracts/polygon/connectors/uniswap/v3_swap/helpers.sol [hunk 7]
```diff
@@ -69,8 +69,8 @@ abstract contract Helpers is DSMath, Basic {
 		uint256 _sellAmt = swapRouter.exactOutputSingle(params);
 		require(_slippageAmt >= _sellAmt, "Too much slippage");
 
-		isMatic = address(_buyAddr) == wmaticAddr;
-		convertWmaticToMatic(isMatic, _buyAddr, _buyAmt);
+		isMatic = address(buyData.buyAddr) == maticAddr;
```
Comment (thrilok209) [inline contracts/polygon/connectors/uniswap/v3_swap/helpers.sol:77]:
```suggestion
                if (_slippageAmt - _sellAmt > 0) convertMaticToWmatic(isMatic, _sellAddr, _slippageAmt - _sellAmt);
		isMatic = address(buyData.buyAddr) == maticAddr;
```

**Comment by pradyuman-verma** [comment]
### Deployed on Poly

Name: `UNISWAP-V3-SWAP-A`
Addr: [0x7D9eaeD9Bb41dC82b9A2Bcd4ed097F0e576C4CBc](https://polygonscan.com/address/0x7D9eaeD9Bb41dC82b9A2Bcd4ed097F0e576C4CBc)
tx: [here](https://polygonscan.com/tx/0xfc2ecafef633c3f018be985227dd80dbfa99cd99bdd7a63136c7f8b49740b1b1)

### Deployed on OPT

Name: `UNISWAP-V3-SWAP-A`
Addr: [0x1bD311bd5aD4DCfE18bCb55d42F0a3c714E2bfDe](https://optimistic.etherscan.io/address/0x1bD311bd5aD4DCfE18bCb55d42F0a3c714E2bfDe#code)
tx: [here](https://optimistic.etherscan.io/tx/0x1d03a61ed37656a4328fd540bb1c2c04f4e5fcf5be50c00fc8cad600cf1b8650)


### Deployed on ARB

Name: `UNISWAP-V3-SWAP-A`
Addr: [0x670B8378D53AeD04cDeBa632C2276F4414D693b2](https://arbiscan.io/address/0x670B8378D53AeD04cDeBa632C2276F4414D693b2#code)
tx: [here](https://arbiscan.io/tx/0x77e26beaa8a8e3772e2107e4d5a211f3c651fd8372d489867a5e89cb76db4f88)


### Deployed on Mainnet

Name: `UNISWAP-V3-SWAP-A`
Addr: [0xDD6f1a8036f8F65a6f53F1818F928D4153A8DaD3](https://etherscan.io/address/0xDD6f1a8036f8F65a6f53F1818F928D4153A8DaD3#code)
tx: [here](https://etherscan.io/tx/0xacc99516209f213f2899651a675b9bb1408df8cbc05095de46a5fee47ead7fb0)

**Review by thrilok209 — APPROVED** [review]

**Pull request merged**